client/app/core/layouts.config.js

```diff
@@ -19,25 +19,31 @@ function getLayouts() {
 }
 
 /** @ngInject */
-function enterApplication(Polling, lodash, $state, NavCounts, Navigation) {
+function enterApplication(Polling, lodash, $state, NavCounts, Navigation, Session, $timeout) {
   // Application layout displays the navigation which might have items that require polling to update the counts
-  angular.forEach(NavCounts.counts, updateCount);
-  angular.forEach(Navigation.items, function(value, key) {
-    lodash.merge(value, $state.navFeatures[key]);
-    angular.forEach(value.secondary, function(secondaryValue, secondaryKey) {
-      lodash.merge(secondaryValue, $state.actionFeatures[secondaryKey]);
-    });
-  });
-
+ 
+  function initBadgeCount() {
+    if (angular.isDefined($state.navFeatures)) {
+      angular.forEach(NavCounts.counts, updateCount);
+      angular.forEach(Navigation.items, function (value, key) {
```

Comment:
Unless you're using [arrow shorthand](http://babeljs.io/learn-es2015/#ecmascript-2015-features-arrows-and-lexical-this)

prefer use of named functions, similar to the way `updateCount` is used on line 27